src/common/chunked.ts

```diff
@@ -0,0 +1,315 @@
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { pLimit } from 'plimit-lit';
+import { Readable } from 'stream';
+
+import type {
+  TurboLogger,
+  TurboUploadDataItemResponse,
+  UploadSignedDataItemParams,
+} from '../types.js';
+import { TurboEventEmitter } from './events.js';
+import { TurboHTTPService } from './http.js';
+
+export const fiveMiB = 5 * 1024 * 1024; // 5 MiB
+export const fiveHundredMiB = fiveMiB * 100; // 500 MiB
+export const defaultChunkSize = fiveMiB; // Default chunk size for uploads
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+/**
+ * Events emitted during chunked upload.
+ */
+export interface ChunkingEvents {
+  /** Fired after each successful chunk upload */
+  chunkUpload: (info: {
+    id: number;
+    offset: number;
+    size: number;
+    totalUploaded: number;
```

Comment:
Nit: add unit in name for clarity (I.e. chunk or bytes)

Reply:
fe35ffd